File: dojo_9/java/src/main/java/com/mercadolibre/dojos/Viaje.java
package com.mercadolibre.dojos;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Viaje {
    private List<Tramo> tramos = new ArrayList();

    public Viaje(Tramo... tramos) {
        this.tramos = Arrays.asList(tramos);
    }

    public Precio precio() {
        Precio precio = this.tramos
                .stream()
                .map(tramo -> tramo.precio())
                .reduce(
                    new Precio(0), (precio1, precio2) -> {
                        return precio1.plus(precio2);
                    });

        return precio;
    }
}

File: dojo_kotlin/app/src/main/java/com/nelsito/kotlindojo9/Viaje.kt
package com.nelsito.kotlindojo9

class Viaje(vararg val tramo: Tramo) {
    fun precio(): Precio {
        return tramo
            .map { t -> t.precio() }
            .reduce {
                precio1, precio2 -> precio1.plus(precio2)
            }
    }
}